src/aggregation/metric/stats.rs

```diff
@@ -81,42 +128,253 @@ impl Stats {
+/// the bounds from the average with a distance of
+/// std_deviation*sigma
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct StandardDeviationBounds {
+    /// upper bound -> avg + std_dev*sigma
+    pub upper: f64,
+    /// lower bound -> avg - std_dev*sigma
+    pub lower: f64,
+    /// upper bound sampling -> avg + std_dev_sampling*sigma
+    pub upper_sampling: f64,
+    /// lower bound sampling -> avg - std_dev_sampling*sigma
+    pub lower_sampling: f64,
+    /// same as upper
+    pub upper_population: f64,
+    /// same as lower
+    pub lower_population: f64,
+}
+
+impl ExtendedStats {
+    pub(crate) fn get_value(&self, agg_property: &str) -> crate::Result<Option<f64>> {
+        match agg_property {
+            "count" => Ok(Some(self.count as f64)),
+            "sum" => Ok(Some(self.sum)),
+            "min" => Ok(self.min),
+            "max" => Ok(self.max),
+            "avg" => Ok(self.avg),
+            "variance" => Ok(self.variance),
+            "variance_sampling" => Ok(self.variance_sampling),
+            "variance_population" => Ok(self.variance_population),
+            "sum_of_squares" => Ok(self.sum_of_squares),
+            "std_deviation" => Ok(self.std_deviation),
+            "std_deviation_sampling" => Ok(self.std_deviation_sampling),
+            "std_deviation_population" => Ok(self.std_deviation_population),
+            "std_deviation_bounds.lower" => Ok(self
+                .std_deviation_bounds
+                .as_ref()
+                .map(|bounds| bounds.lower)),
+            "std_deviation_bounds.lower_population" => Ok(self
+                .std_deviation_bounds
+                .as_ref()
+                .map(|bounds| bounds.lower_population)),
+            "std_deviation_bounds.lower_sampling" => Ok(self
+                .std_deviation_bounds
+                .as_ref()
+                .map(|bounds| bounds.lower_sampling)),
+            "std_deviation_bounds.upper" => Ok(self
+                .std_deviation_bounds
+                .as_ref()
+                .map(|bounds| bounds.upper)),
+            "std_deviation_bounds.upper_population" => Ok(self
+                .std_deviation_bounds
+                .as_ref()
+                .map(|bounds| bounds.upper_population)),
+            "std_deviation_bounds.upper_sampling" => Ok(self
+                .std_deviation_bounds
+                .as_ref()
+                .map(|bounds| bounds.upper_sampling)),
+            _ => Err(TantivyError::InvalidArgument(format!(
+                "Unknown property {agg_property} on stats metric aggregation"
+            ))),
+        }
+    }
+}
+
 /// Intermediate result of the stats aggregation that can be combined with other intermediate
 /// results.
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+
+#[derive(Default, Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct IntermediateStats {
+    stats: IntermediateExtendedStats,
```

Comment:
I think the memory overhead for e.g. terms -> avg aggregation is too high

Performance may also be impacted
`cargo +nightly bench --features unstable aggregation::agg_bench`